cmake/modules/CheckCxxAtomic.cmake

```diff
@@ -62,7 +62,7 @@ if(NOT HAVE_CXX11_ATOMIC)
   check_cxx_atomics(HAVE_LIBATOMIC)
   cmake_pop_check_state()
   if(HAVE_LIBATOMIC)
-    set(LIBATOMIC_LINK_FLAGS "-Wl,--as-needed -latomic")
+    set(LIBATOMIC_LINK_FLAGS "-Wl,--as-needed -latomic -lmlx5 -libverbs -lrdmacm")
```

Comment:
these compilation options have nothing to do with libatomic. if rdma requires ibverbs and rdmacm, i'd suggest link against `IBVerbs::verbs` and `RDMA::RDMAcm` respectively.

Reply:
When the rdma compilation option is turned on in BuildSPDK.cmake (eg, --with-rdma=mlx5_dv),ceph will report an error when compiling spdk-related files. Need to add the three compilation options of lmlx5, libverbs and lrdmacm for the compilation to pass.

Reply:
> these compilation options have nothing to do with libatomic. if rdma requires ibverbs and rdmacm, i'd suggest link against `IBVerbs::verbs` and `RDMA::RDMAcm` respectively.

When the rdma compilation option is turned on in BuildSPDK.cmake (eg, --with-rdma=mlx5_dv),ceph will report an error when compiling spdk-related files. Need to add the three compilation options of lmlx5, libverbs and lrdmacm for the compilation to pass.

Reply:
@lreeze123 again, i am not against fixing the issue. but 

1. again, please link these libraries at the appropriate place
2. please amend your commit message to explain the issue

and, if you modify `BuildSPDK.cmake` directly to enable SPDK to build the RDMA transport, and we end up with a linker error, i think we should fix the SPDK build instead, not Ceph.

Reply:
Ok, I will revise it and submit again.